### PR TITLE
EDG-13557: GET method

### PR DESCRIPTION
**Pull request by francisf2**
Implement GET method for S3 and local directory services

### CODE REVIEW

**Review by leshow — APPROVED** [review]

**Comment by leshow** [comment]
Looks fine to me, you may want to add some description in the PR though

File: aws.go [hunk 1]
```diff
@@ -39,13 +40,34 @@ type AwsPlugin struct {
 }
 
 func (plugin AwsPlugin) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+	switch req.Method {
+	case "PUT":
```
Comment (abrener5735) [inline aws.go:44]: Should use http.MethodPut and http.MethodGet

File: aws.go [hunk 2]
```diff
@@ -39,13 +40,34 @@ type AwsPlugin struct {
...
 	resp, err := plugin.service.Put(req.URL.Path[1:], payload, req.Header.Get("Content-Type"), rw)
+	handleResponse(resp, err, rw)
+}
+
+func (plugin *AwsPlugin) get(rw http.ResponseWriter, req *http.Request) {
+	resp, err := plugin.service.Get(req.URL.Path[1:], rw)
+	handleResponse(resp, err, rw)
+}
+
+func handleResponse(resp []byte, err error, rw http.ResponseWriter) {
```
Comment (abrener5735) [inline aws.go:70]: Can be confusing to use the name `err` as an argument and internal variable. I would rename the argument, maybe to `requestErr`

File: s3/s3.go [hunk 3]
```diff
@@ -35,9 +35,13 @@ func New(bucket, prefix, region string, timeoutSeconds int, creds *ecs.Credentia
 	}
 }
 
-func (s3 *S3) Put(name string, payload []byte, contentType string, rw http.ResponseWriter) ([]byte, error) {
+func (s3 *S3) request(httpMethod string, name string, payload []byte, contentType string, rw http.ResponseWriter) ([]byte, error) {
 	uri := s3.bucketUri + s3.prefix + "/" + name
-	req, err := http.NewRequest("PUT", uri, bytes.NewReader(payload))
+	var payloadReader io.Reader = nil
+	if payload != nil {
+		payloadReader = bytes.NewReader(payload)
```
Comment (abrener5735) [inline s3/s3.go:42]: Given that the payload may be big, it may be more efficient to use `bufio.NewReader(..)`

File: s3/s3.go [hunk 4]
```diff
@@ -66,6 +72,14 @@ func (s3 *S3) Put(name string, payload []byte, contentType string, rw http.Respo
 	return response, nil
 }
 
+func (s3 *S3) Put(name string, payload []byte, contentType string, rw http.ResponseWriter) ([]byte, error) {
+	return s3.request("PUT", name, payload, contentType, rw)
```
Comment (abrener5735) [inline s3/s3.go:76]: Should use `http.MethodPut` and `http.MethodGet`

**Pull request merged**